### PR TITLE
Feature/Improve base components to fit all apps need better

### PR DESCRIPTION
**Pull request by mateuszRybczonek**

### CODE REVIEW

File: src/components/BaseModal.vue [hunk 1]
```diff
@@ -2,33 +2,33 @@
   <Transition>
     <div
       v-if="isVisible"
-      class="base-modal-overlay"
+      class="base-modal__overlay"
```
Comment (Paterick) [inline src/components/BaseModal.vue:5]: I'm not sure I understand when we are using __ in the class name vs. hyphen (-) is this a rule I should be following?

Comment (mateuszRybczonek) [reply src/components/BaseModal.vue:5]: I think it is easier to read if we use BEM methodology.

http://getbem.com/naming/

It makes it more readable, what I usually follow is:
- every class in a component starts with some meaningful name, usually the component name (that makes it much easier to find a component later when you look at the html to debug something)
- then inside the component I use the `element` with `__` prefix so my BaseModal has an `overlay`, `content` etc.
- then if there is any additional modifier like `disabled` we use `--` prefix, that allows us to see that it is the same element but in a very specific state

**Review by Paterick — APPROVED** [review]

File: src/components/BaseButton.vue [hunk 2]
```diff
@@ -81,9 +81,36 @@ const content = ref()
...
 const kButtonTypeClass = {
   [EButtonType.primary]: 'primary',
-  [EButtonType.secondary]: 'secondary'
+  [EButtonType.secondary]: 'secondary',
+  [EButtonType.tertiary]: 'tertiary',
+  [EButtonType.gray]: 'gray'
 }
 
+const kSpinnerColor = {
+	[EButtonType.primary]: {
```
Comment (robbiecarlton) [inline src/components/BaseButton.vue:90]: Tabs thats should be spaces here, and in the styles below

File: src/components/BaseInput.vue [hunk 3]
```diff
@@ -1,10 +1,25 @@
 <template>
   <div
     class="base-input"
-    :class="{ 'disabled' : isDisabled }"
+    :class="{ 'base-input--disabled' : isDisabled, 'base-input--is-horizontal': isHorizontal }"
     data-test-base-input-wrapper
   >
-    <div>
+		<slot
```
Comment (robbiecarlton) [inline src/components/BaseInput.vue:7]: tabs

File: src/components/BaseModal.vue [hunk 4]
```diff
@@ -93,90 +80,103 @@ const emit = defineEmits(['close'])
...
-
-.base-modal-close-button {
-  position: absolute;
-  top: 10px;
-  right: 10px;
-  border: none;
-  padding: 0;
-  background-color: transparent;
-  cursor: pointer;
+	&__overlay {
```
Comment (robbiecarlton) [inline src/components/BaseModal.vue:84]: woah, since when can we use `&`?

File: src/components/BaseModal.vue [hunk 5]
```diff
@@ -93,90 +80,103 @@ const emit = defineEmits(['close'])
...
+		position: fixed;
+		display: flex;
```
Comment (robbiecarlton) [inline src/components/BaseModal.vue:86]: tabs in this file

**Review by robbiecarlton — APPROVED** [review]
couple of cleanup comments, otherwise looks good. Good stuff!

**Pull request merged**